### PR TITLE
#406 does not work for 3D data

### PR DESCRIPTION
**Pull request by jkotan**
Without the PR on gets the following error for RGB data

```
Traceback (most recent call last):
  File "/home/jkotan/sources/lavue/lavuelib/imageDisplayWidget.py", line 64, in paint
    _pg.ImageItem.paint(self, p, *args)
  File "/home/jkotan/testinst/lib/python2.7/site-packages/pyqtgraph/graphicsItems/ImageItem.py", line 439, in paint
    self.render()
  File "/home/jkotan/testinst/lib/python2.7/site-packages/pyqtgraph/graphicsItems/ImageItem.py", line 431, in render
    argb, alpha = fn.makeARGB(image, lut=lut, levels=levels)
  File "/home/jkotan/testinst/lib/python2.7/site-packages/pyqtgraph/functions.py", line 1187, in makeARGB
    imgData[nanMask, 3] = 0
IndexError: too many indices for array
```

### CODE REVIEW

File: pyqtgraph/functions.py [hunk 1]
```diff
@@ -1111,7 +1111,7 @@ def makeARGB(data, lut=None, levels=None, scale=None, useRGBA=False):
     # awkward, but fastest numpy native nan evaluation
     # 
     nanMask = None
-    if data.dtype.kind == 'f' and np.isnan(data.min()):
+    if data.ndim == 2 and data.dtype.kind == 'f' and np.isnan(data.min()):
```
Comment (ixjlyons) [inline pyqtgraph/functions.py]: For a given pixel where one or more of the color channel values is nan, the color of that pixel is kind of "undefined", so making it transparent makes the most sense I think. We could collapse the nan mask along the color channel axis and still use it:

```suggestion
    if data.dtype.kind == 'f' and np.isnan(data.min()):
        nanMask = np.isnan(data)
        if data.ndim > 2:
            nanMask = np.any(nanMask, axis=data.ndim-1)
```

File: pyqtgraph/functions.py [hunk 2]
```diff
@@ -1113,6 +1113,9 @@ def makeARGB(data, lut=None, levels=None, scale=None, useRGBA=False):
     nanMask = None
     if data.dtype.kind == 'f' and np.isnan(data.min()):
         nanMask = np.isnan(data)
+        if data.ndim > 2:
+            nanMask = np.any(nanMask, axis=-1)
+        nanMask = np.isnan(data)
```
Comment (ixjlyons) [inline pyqtgraph/functions.py]: This line should be removed

**Comment by j9ac9k** [comment]
there is a merge conflict, I'll see if I can resolve....

**Comment by j9ac9k** [comment]
Contents of this PR were merged as part of #1152 thanks for the PR @jkotan and thanks for the tests @ixjlyons

**Pull request closed without merging**